sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -307,6 +307,7 @@ case class MapPartitions[T, U](
     uEncoder: ExpressionEncoder[U],
     output: Seq[Attribute],
     child: SparkPlan) extends UnaryNode {
+  override def missingInput: AttributeSet = AttributeSet.empty
```

Comment:
Is the `!` prefix the only influence if we have `missingInput` in `SparkPlan`? There is no `resolved` concept for `SparkPlan` and it looks to me that physical plan doesn't need to care about these stuff.